Object values (#7484)
Codecademy/codepedia-content: content/javascript/concepts/objects/terms/values/values.md

---
Title: '.values()'
Description: 'Returns an array containing the values of an object’s own enumerable properties.'
Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'JavaScript'
  - 'Methods'
  - 'Objects'
  - 'Properties'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

> The **`Object.values()`** method returns an array containing the values of an object's own enumerable properties. It's useful when you only need the values, not the keys.

Reviewer: ```suggestion
The **`Object.values()`** method returns an array containing the values of an object's own enumerable properties. This method is useful when only the values are needed, not the keys.
```

**Return value:**

- An array of values corresponding to the object's own enumerable properties, ordered according to JavaScript's property ordering rules (integer keys first, then string keys in insertion order).
- An array of the object’s own enumerable property values, ordered by integer keys first (ascending), then string keys in insertion order.

## Example 1: Basic usage of `Object.values()`

In this example, `Object.values()` creates an array of property values from an object:

```js

> const perfectWife = {

Reviewer 2: Hey @Ariel-GonzAguer, the object `perfectWife` sounded a bit vague. So, to keep the language professional, just changed the object name 😅

## Codebyte Example: `Object.values()` with `.filter()` method to detect booleans in a config object

In this example, `Object.values()` is combined with `.filter()` to extract only boolean values from an object:

```codebyte/javascript
const catObject = {
  ears = true,
  nose = "little",
  beans = true,
  wings = false,
  tails = 1,
  ears: true,
  nose: "little",
  beans: true,
  wings: false,
  tails: 1,
}

const booleanValues = Object.values(catObject).filter((value) => typeof value === "boolean");

console.log(booleanValues);
```

> ## Frequently Asked Questions

Reviewer 2: So, ideally FAQs are kind of a test that we are doing. And you'll see only the maintainers adding FAQs to the PRs, and not the authors. that's why I removed them. Hope this helps! 😄

> ## Frequently Asked Questions

Reviewer 2: @Ariel-GonzAguer I discussed this with the team yesterday - they said the Authors can add FAQs, just take care of these points:
- FAQs should be the common queries asked by learners
- You can get these FAQs from the "People also ask" section on google, here is an example:
<img width="1487" height="892" alt="image" src="https://github.com/user-attachments/assets/169b1dbe-d11e-483c-ba78-8027e536cd2b" />

So, while adding FAQs, focus on the top questions. I hope this helps 😄
